### PR TITLE
Add descendant selectors

### PR DESCRIPTION
**Pull request by glyn**
Add array slicing, filters, and lists.

Fixes https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/219

### CODE REVIEW

File: draft-ietf-jsonpath-base.md [hunk 1]
```diff
@@ -1047,13 +1048,18 @@ Queries:
 
 The descendant selectors start with a double dot `..` and can be
 followed by an object member name (similar to the `dot-selector`),
-by an `index-selector` acting on objects or arrays, or by a wildcard.
+a wilcard (similar to the `dot-wild-selector`),
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:1051]: wildcard

Comment (cabo) [inline draft-ietf-jsonpath-base.md:1051]:
```suggestion
a wildcard (similar to the `dot-wild-selector`),
```

**Comment by cabo** [comment]
Would it be useful to have references to indications of implementation consensus in this ticket?

**Review by gregsdennis — APPROVED** [review]

**Comment by glyn** [comment]
> Would it be useful to have references to indications of implementation consensus in this ticket?

Probably best to put such things in issues rather than PRs, so I'll add something to https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/219.

**Pull request merged**